src/c2goto/CMakeLists.txt

```diff
@@ -38,19 +38,21 @@ endif()
 
 function(mangle_clib output)
   set(result "")
-   file(GLOB c2goto_library_files CONFIGURE_DEPENDS
+  # Extract a list of C's operational models
+  file(GLOB c2goto_library_files CONFIGURE_DEPENDS
     "${CMAKE_CURRENT_SOURCE_DIR}/library/*.c")
+  # Extract a list of C's math operational models
   file(GLOB c2goto_libm_files CONFIGURE_DEPENDS
     "${CMAKE_CURRENT_SOURCE_DIR}/library/libm/*.c")
   file(GLOB_RECURSE c2goto_header_files CONFIGURE_DEPENDS
     "${CMAKE_CURRENT_SOURCE_DIR}/headers/*.h")
+
+  # For each config, generate clib*.c and clib*.goto file in the current binary directory
   foreach(in_f ${ARGN})
     set (in_f "${${in_f}}")
-    list(GET in_f 0 in_file)
+    list(GET in_f 0 out_file)
     list(SUBLIST in_f 1 -1 in_flags)
-    file(RELATIVE_PATH out_file ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_SOURCE_DIR}/${in_file})
-    string(REGEX REPLACE .c .goto out_goto "${in_file}")
-    string(REGEX REPLACE .c "" barename "${in_file}")
+    string(REGEX REPLACE .c .goto out_goto "${out_file}")
```

Comment:
Could we use `string(REPLACE .c .goto [...])` here instead of `REGEX REPLACE` since the dot `.` has special regex meaning?

Reply:
Done